grakn/rpc/session.py

```diff
@@ -55,10 +56,9 @@ def __init__(self, client, database: str, session_type: SessionType, options=Gra
 
         self._session_id = self._grpc_stub.session_open(open_req).session_id
         self._is_open = True
-        self._pulse = self._scheduler.enter(delay=self._PULSE_FREQUENCY_SECONDS, priority=1, action=self._transmit_pulse, argument=())
-        # TODO: This thread blocks the process from closing. We should try cancelling the scheduled task when the
-        #       session closes. If that doesn't work, we need some other way of getting the thread to exit.
-        Thread(target=self._scheduler.run, daemon=True).start()
+        self._pulse_thread_pool = ThreadPoolExecutor(thread_name_prefix='session_pulse_{}'.format(self._session_id.hex()))
```

Comment:
Suppose a user has 8 sessions open in parallel; then this will create 8 separate `ThreadPoolExecutor`s, right? Do you think there's any benefit to be gained from having a single `ThreadPoolExecutor` in the `GraknClient` and reusing that across all sessions?